Dockerfile.skaffold

```diff
@@ -1,4 +1,4 @@
-ARG GO_VERSION=1.19.9
+ARG GO_VERSION=1.20.5
```

Comment:
This file needs to be added to the golang bump action in https://github.com/elastic/elastic-agent/blob/98327dd5292923c2fa919f2671a357439a3d5ced/.ci/bump-golang.yml#L87